Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device.cpp

```diff
@@ -2,47 +2,137 @@
 // Licensed under GPLv2+
 // Refer to the license.txt file included.
 
-#include "Core/IPC_HLE/WII_IPC_HLE.h"
+#include <algorithm>
+
 #include "Common/StringUtil.h"
 #include "Core/HW/Memmap.h"
 #include "Core/HW/SystemTimers.h"
+#include "Core/IPC_HLE/WII_IPC_HLE.h"
 #include "Core/IPC_HLE/WII_IPC_HLE_Device.h"
 
-SIOCtlVBuffer::SIOCtlVBuffer(const u32 address) : m_Address(address)
+IOSResourceRequest::IOSResourceRequest(const u32 command_address) : address(command_address)
+{
+  command = static_cast<IPCCommandType>(Memory::Read_U32(command_address));  // 0x00
+  return_value = Memory::Read_U32(command_address + 4);                      // 0x04
+  fd = Memory::Read_U32(command_address + 8);                                // 0x08
+}
+
+void IOSResourceRequest::SetReturnValue(const s32 new_return_value)
+{
+  return_value = new_return_value;
+  Memory::Write_U32(return_value, address + 4);
```

Comment:
should have a static_cast on return_value here, since it's a signed value, whereas Write_U32 takes a u32.